### PR TITLE
Feature/#312 CFEI Filter Views + API Integration

### PR DESCRIPTION
**Pull request by kenzo23312**
CFEI Filter Views + API Integration

### CODE REVIEW

File: frontend/src/components/eois/filters/eoiDsFilter.js [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+  componentWillMount() {
+    const { pathName, query } = this.props;
+    resetChanges(pathName, query);
+  }
+
+  onSearch(values) {
+    const { pathName, query } = this.props;
+
+    const { title } = values;
```
Comment (marcindo) [inline frontend/src/components/eois/filters/eoiDsFilter.js]: you can combine all this assignments into one object

File: frontend/src/components/eois/filters/eoiDsFilter.js [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    const { agency } = values;
+    const { active } = values;
+    const { country_code } = values;
+    const { specializations } = values;
+    const { selected_source } = values;
+
+    history.push({
+      pathname: pathName,
+      query: R.merge(query, {
+        title, agency, active, country_code, specializations, selected_source,
```
Comment (marcindo) [inline frontend/src/components/eois/filters/eoiDsFilter.js]: you probably don't need destructure here at all, but just merge current query with values

File: frontend/src/components/eois/filters/eoiDsFilter.js [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+      }),
+    });
+  }
+
+  render() {
+    const { classes, countries, directSources, specs, handleSubmit, reset } = this.props;
+
+    return (
+      <form onSubmit={handleSubmit(this.onSearch)}>
```
Comment (marcindo) [inline frontend/src/components/eois/filters/eoiDsFilter.js]: there should be common file with filter fields that appear in other cfei filters, as I'm sure they have exact same props

File: frontend/src/components/eois/filters/eoiDsFilter.js [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+
+    this.onSearch = this.onSearch.bind(this);
+  }
+
+  componentWillMount() {
+    const { pathName, query } = this.props;
+    resetChanges(pathName, query);
+  }
+
+  onSearch(values) {
```
Comment (marcindo) [inline frontend/src/components/eois/filters/eoiDsFilter.js]: basic filter functionality is just component that has search and clear buttons, filter form inside and one function to submit filter by adding it to query, which is always the same, just merge current query with new values, doesn't matter what's inside.
So it should be a common component which accepts children (fields) to render inside form

File: frontend/src/components/common/list/paginatedList.js [hunk 5]
```diff
@@ -63,8 +63,19 @@ class PaginatedList extends Component {
     </Typography></div>);
   }
 
+  changeSorting(sorting) {
```
Comment (marcindo) [inline frontend/src/components/common/list/paginatedList.js]: can you move sorting to separate wrapper that wraps base list and provides changeSorting and sorting props? So we can easily plug in different one

File: frontend/src/components/eois/filters/eoiHelper.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
+import { browserHistory as history } from 'react-router';
+import R from 'ramda';
+
+const resetChanges = (pathName, query) => {
```
Comment (marcindo) [inline frontend/src/components/eois/filters/eoiHelper.js]: instead read current query, make new object with only sorting, page and page size fields and push that, so this function can be reused for all filters

File: frontend/src/components/forms/fields/projectFields/adminOneLocations.js [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    if (this.props.countryCode !== nextProps.countryCode && nextProps.countryCode) {
+      this.setState({ contryCode: nextProps.countryCode });
+      this.props.loadAdminOneLocations({ country_code: nextProps.countryCode });
+    }
+  }
+
+  render() {
+    const { fieldName, label, locations, ...other } = this.props;
+    return (
+      <SelectForm
```
Comment (marcindo) [inline frontend/src/components/forms/fields/projectFields/adminOneLocations.js]: should be disabled if no country was chosen

Comment (kenzo23312) [reply frontend/src/components/forms/fields/projectFields/adminOneLocations.js]: done

File: frontend/src/components/eois/filters/eoiHelper.js (hunk 6 above)
Comment (kenzo23312) [reply frontend/src/components/eois/filters/eoiHelper.js]: done

File: frontend/src/components/eois/filters/eoiDsFilter.js (hunk 2 above)
Comment (kenzo23312) [reply frontend/src/components/eois/filters/eoiDsFilter.js]: done

File: frontend/src/components/eois/filters/eoiDsFilter.js (hunk 1 above)
Comment (kenzo23312) [reply frontend/src/components/eois/filters/eoiDsFilter.js]: done

**Pull request merged**